client/src/main/scala/skuber/api/package.scala

```diff
@@ -51,11 +49,66 @@ package object client {
 
    case class Context(
      cluster: Cluster = Cluster(),
-     authInfo: AuthInfo = AuthInfo(),
+     authInfo: AuthInfo = NoAuth,
      namespace: Namespace = Namespace.default
    )
 
-   case class AuthInfo(
+   sealed trait AuthInfo
+
+   object NoAuth extends AuthInfo {
+     override def toString: String = "NoAuth"
+   }
+
+   final case class BasicAuth(userName: String, password: String) extends AuthInfo {
+     override def toString: String = s"${getClass.getSimpleName}(userName=$userName,password=<redacted>)"
+   }
+
+   final case class TokenAuth(token: String) extends AuthInfo {
+     override def toString: String = s"${getClass.getSimpleName}(token=<redacted>)"
+   }
+
+   final case class CertAuth(clientCertificate: PathOrData, clientKey: PathOrData, user: Option[String]) extends AuthInfo {
+     override def toString: String = StringBuilder.newBuilder
+       .append(getClass.getSimpleName)
+       .append("(")
+       .append {
+         clientCertificate match {
+           case Left(certPath: String) => "clientCertificate=" + certPath + " "
+           case Right(_) => "clientCertificate=<PEM masked> "
+         }
+       }
+       .append {
+         clientKey match {
+           case Left(certPath: String) => "clientKey=" + certPath + " "
+           case Right(_) => "clientKey=<PEM masked> "
+         }
+       }
+       .append("userName=")
+       .append(user.getOrElse(""))
+       .append(" )")
+       .mkString
+   }
+
+   sealed trait AuthProviderAuth extends AuthInfo {
+     def name: String
+   }
+
+   // 'jwt' supports an oidc id token per https://kubernetes.io/docs/admin/authentication/#option-1---oidc-authenticator
+   // - but does not yet support token refresh
+   final case class OidcAuth(idToken: String) extends AuthProviderAuth {
+     override val name = "oidc"
+
+     override def toString = """OidcAuth(idToken=<redacted>)"""
+   }
+
+   final case class GcpAuth(accessToken: String) extends AuthProviderAuth {
+     override val name = "gcp"
+
+     override def toString =
+       """GcpAuth(accessToken=<redacted>)""".stripMargin
+   }
+
+   case class AuthInfo1(
```

Comment:
I assume `AuthInfo1` should be removed?

Reply:
Yeah, I kept that to have easy the old code, should have removed before committing